### DISCUSSION TITLE
Is it possible to refractor types with python abstract class

### DISCUSSION THREAD

**Discussion by ccsv** [strawberry-graphql/strawberry#2314] [Q&A]
Can the abstract class be used to write less code?

For example is it possible to do something like this:

```
from abc import ABC

class Contact(ABC):
    location: str 
    description: str
    phone: int

@strawberry.type
class Home(Address):
    name: str 

@strawberry.type
class Business(Address):
    name: str 
    fax: int
```

Instead of:

```
@strawberry.type
class Home(Address):
    name: str 
    location: str 
    description: str
    phone: int

@strawberry.type
class Business(Address):
    name: str 
    fax: int
    location: str 
    description: str
    phone: int
```

**Comment by ViniciusMarchi** [comment]
Hi. According to your problem I believe that the use of `interfaces` can solve it. Strawberry provides this feature.

Following your example, we have the following implementation using interfaces
```
@strawberry.interface
class Address:
    location: str 
    description: str
    phone: int

@strawberry.type
class Home(Address):
    name: str 

@strawberry.type
class Business(Address):
    name: str 
    fax: int
```
In this way both  `Home` and `Business` will have the fields `location`, `description` and `phone`, defined in the `Address` interface.
For more information check the oficial strawberry [interface documentation](https://strawberry.rocks/docs/types/interfaces).
